packages/world/src/utils/InventoryUtils.sol

```diff
@@ -257,8 +242,21 @@ library InventoryUtils {
 
       InventorySlotData memory sourceSlot = InventorySlot._get(from, slotFrom);
       require(!sourceSlot.objectType.isNull(), "Empty slot");
+      fromSlotData[i] = SlotData(sourceSlot.entityId, sourceSlot.objectType, sourceSlot.amount);
 
       InventorySlotData memory destSlot = InventorySlot._get(to, slotTo);
+
+      // Handle slot swaps (transferring all to an existing slot with a different type)
```

Comment:
What happens right now if you try to combine two slots? one slot with 30 dirt, another with 10 dirt. Would it become one slot with 40 dirt? It wouldnt swap I think cuz you're checking if they're diff object types

Reply:
it would just follow the regular transfer flow from one slot to another, but will add a test